linode/lke/cluster.go

```diff
@@ -489,3 +489,40 @@ func expandLKEClusterControlPlane(controlPlane map[string]interface{}) linodego.
 
 	return result
 }
+
+func removeExternalPools(ctx context.Context, externalPoolTags []string, pools []linodego.LKENodePool) []linodego.LKENodePool {
+	var filteredPools []linodego.LKENodePool
+
+	if len(externalPoolTags) == 0 {
+		return pools
+	}
+
+	tagSet := make(map[string]struct{}, len(externalPoolTags))
+	for _, tag := range externalPoolTags {
+		tagSet[tag] = struct{}{}
+	}
+
+	for _, pool := range pools {
+		tag := poolHasAnyOfTags(pool, tagSet)
+		if tag == "" {
+			filteredPools = append(filteredPools, pool)
+		} else {
+			tflog.Info(ctx, "Excluding pool from management by this resource", map[string]interface{}{
+				"pool_id": pool.ID,
+				"tag":     tag,
+				"reason":  "Pool tagged to be managed by a separate linode_lke_node_pool resource",
+			})
+		}
+	}
+
+	return filteredPools
+}
+
+func poolHasAnyOfTags(pool linodego.LKENodePool, tagSet map[string]struct{}) string {
+	for _, poolTag := range pool.Tags {
+		if _, exists := tagSet[poolTag]; exists {
+			return poolTag
+		}
+	}
+	return ""
+}
```

Comment:
nitpick: It might make sense to return a pointer here so we can compare against `null` rather than an empty string

```suggestion
func poolHasAnyOfTags(pool linodego.LKENodePool, tagSet map[string]any) *string {
	for _, poolTag := range pool.Tags {
		if _, exists := tagSet[poolTag]; exists {
			result := poolTag

			return &result
		}
	}
	return nil
}
```

Reply:
done